File: Server/routes/UserRoutes.js
const express = require('express');
const router = express.Router();
const userController = require('../controllers/userController.js');
const isAuthenticated = require('../Utils/AuthenticationUtil.js'); 

router.get('/info', isAuthenticated, userController.getUser);

router.post('/create', isAuthenticated, userController.createUser);

router.get('/get', isAuthenticated, userController.getUserByAttribute);

router.post('/update', isAuthenticated, userController.updateUser);

router.post('/register', userController.register);

router.post('/reset', userController.reset);

router.post('/validate', userController.validateUser);

router.post('/login', userController.authorizeLogin);

router.get('/logout', userController.logout);

router.get('/access', isAuthenticated, userController.access);

module.exports = router;

File: Server_Go/routes/UserRoutes.go
package userRoutes

import (
	"github.com/gorilla/mux"
)

func RegisterUserRoutes(r *mux.Router) {
	r.HandleFunc("/info", userController.getUser).Name("User Info")
	r.HandleFunc("/create", userController.createUser).Name("Create User")
	r.HandleFunc("/get", userController.getUserByAttribute).Name("Get User")
	r.HandleFunc("/update", userController.updateUser).Name("Update User")
	r.HandleFunc("/register", userController.register).Name("Register User")
	r.HandleFunc("/reset", userController.reset).Name("Reset User")
	r.HandleFunc("/validate", userController.validateUser).Name("Validate User")
	r.HandleFunc("/login", userController.authorizeLogin).Name("Authorize User")
	r.HandleFunc("/logout", userController.logOut).Name("Logout User")
	r.HandleFunc("/access", userController.access).Name("User Access")
}
